relayer/src/cache.rs

```diff
@@ -63,6 +76,9 @@ impl Cache {
         }
     }
 
+    /// Attempts to fetch a [`ChannelEnd`] via its [`PortChannelId`]. If a cache miss
+    /// is encountered, this method attempts to insert the channel into the cache,
+    /// assuming the returned channel state is open.
```

Comment:
It is more like: return a cached result in case there is a cache hit for the given `PortChannelId`. Otherwise attempt to fetch the `ChannelEnd` using the provided fetcher function `f`. If the fetcher returns an `Ok` result with the channel end in open state, store a copy of it in the cache and then return the result.

Similar applies to the other functions.

Reply:
Nitpicking, but agree that this comment could use some rephrasing. Some questions to understand possible confusions that could arise:

- "Attempts to fetch" is ambiguous: fetch from cache or fetch from chain via `f`?
    - could restate as "Attempts to read from cache" or Soares' proposal
- "attempts to insert the channel into the cache": maybe document that this method resorts to an expensive call to the chain (via `f`) upon cache miss.
- "assuming the returned channel state is open": it is unclear where is the channel returned from? the channel returned from `f` or from this method?